src/views/FlexExampleAppHomeView.mxml

```diff
@@ -1,204 +1,129 @@
-			});
-			
-			VideoAd.getInstance().addEventListener(HeyzapEvent.ON_AUDIO_FINISHED, function(event:HeyzapEvent=null) {
-				trace("[VIDEO] ON_AUDIO_FINISHED");
-				return null;
-			});
-			
-			// INCENTIVIZED CALLBACKS
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.DID_SHOW, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] DID_SHOW");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.DID_CLICK, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] DID_CLICK");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.DID_HIDE, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] DID_HIDE");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.DID_FAIL, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] DID_FAIL");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.ON_AUDIO_STARTED, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] ON_AUDIO_STARTED");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.ON_AUDIO_FINISHED, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] ON_AUDIO_FINISHED");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.IS_AVAILABLE, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] IS_AVAILABLE");
-				if (IncentivizedAd.getInstance().isAvailable()) {
-					trace("Incentivized Ad is available.");
-				} else {
-					trace("Incentivized Ad is not available");
-				}
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.FETCH_FAILED, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] FETCH_FAILED");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.IS_COMPLETED, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] IS_COMPLETED");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.IS_INCOMPLETE, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] IS_INCOMPLETE");
-				return null;
-			});
-			
-		]]>
+        import com.heyzap.sdk.ads.BannerAd;
+        import com.heyzap.sdk.ads.HeyzapAds;
+        import com.heyzap.sdk.ads.IncentivizedAd;
+        import com.heyzap.sdk.ads.InterstitialAd;
+        import com.heyzap.sdk.ads.VideoAd;
+
+        import model.AppModel;
+
+        [Bindable]
+        public var appModel:AppModel;
+
+        private var currentActionObject:*;
+
+        override public function set data(value:Object):void {
+            appModel = value as AppModel;
+        }
+
+        private function onCreationComplete():void {
+            currentActionObject = InterstitialAd.getInstance();
+        }
+
+        private function getEnteredAdTagOrNull():String{
+            if (!adTagTextInput.text) return null;
```

Comment:
does this return `null` when the tag text field is empty? that'd be the desired behavior.